src/mxarray.jl

```diff
@@ -406,13 +406,13 @@ end
 # char arrays and string
 
 function mxarray(s::String)
-    utf16string = transcode(UInt16, s)
+    utf16string = transcode(mxChar, s)
```

Comment:
we are marshalling it to matlab  using `_mx_create_char_array` in the next line and matlab 
http://www.mathworks.com/help/matlab/apiref/mxchar.html

also thread: https://github.com/JuliaInterop/MATLAB.jl/pull/76